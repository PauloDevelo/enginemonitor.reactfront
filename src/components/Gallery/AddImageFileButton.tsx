import React from 'react';
import {Button} from 'reactstrap';

import { faImage, faPlus } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import { ImageModel } from '../../types/Types';

import {resizeAndSaveImage} from '../../helpers/ImageHelper';

import errorService from '../../services/ErrorService';

type Props = {
    parentUiId: string,
    className?: string,
    addImage: (image: ImageModel) => void
}

const createFileSelector = (onChange: (e: Event)=>void):HTMLInputElement => {
    const fileSelector = document.createElement('input');
    fileSelector.setAttribute('type', 'file');
    fileSelector.addEventListener('change', onChange);

    return fileSelector;
}

function AddImageFileButton({parentUiId, addImage, className}: Props){
    const onFileSelected = (file: File) => {
        uploadImageFile(file, "multer");
    };

    const uploadImageFile = async(file: File, method: string) => {
        try{
            let newImage:ImageModel | undefined = undefined;

            if(method === "multer"){
                newImage = await resizeAndSaveImage(file, parentUiId);
                addImage(newImage);
            }
        }
        catch(error){
            errorService.addError(error);
        }
    };

    const onChange = (e: Event) => {
        const changeEvent = e as unknown as React.ChangeEvent<HTMLInputElement>;
        const files = changeEvent.target.files ? changeEvent.target.files : new FileList() ;
        
        if(files.length > 0){
            onFileSelected(files[0]);
        }
    }

    const fileSelector = createFileSelector(onChange);

    const handleFileSelect = (e: React.MouseEvent):void => {
        e.preventDefault();
        fileSelector.click();
    }

    return (
    <Button color="light" size="lg" className={className} onClick={handleFileSelect} aria-label="Edit">
        <span className="fa-layers fa-fw">
            <FontAwesomeIcon icon={faImage} size="lg"/>
            <FontAwesomeIcon icon={faPlus} size="xs" transform="down-13 left-16"/>
        </span>
    </Button>);
}

export default React.memo(AddImageFileButton);