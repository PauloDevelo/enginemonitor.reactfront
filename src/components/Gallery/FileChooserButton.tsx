import React from 'react';
import {Button} from 'reactstrap';

import { faImage, faPlus } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

type Props = {
    className?: string;
    onFileSelect: (file: File) => any;
}

const createFileSelector = (onChange: (e: Event)=>void):HTMLInputElement => {
    const fileSelector = document.createElement('input');
    fileSelector.setAttribute('type', 'file');
    fileSelector.addEventListener('change', onChange);

    return fileSelector;
}

function FileChooserButton({onFileSelect, className}: Props){
    const onChange = (e: Event) => {
        const changeEvent = e as unknown as React.ChangeEvent<HTMLInputElement>;
        const files = changeEvent.target.files ? changeEvent.target.files : new FileList() ;
        
        if(files.length > 0){
            onFileSelect(files[0]);
        }
    }

    const fileSelector = createFileSelector(onChange);

    const handleFileSelect = (e: React.MouseEvent):void => {
        e.preventDefault();
        fileSelector.click();
    }

    return (
    <Button color="light" size="lg" className={className} onClick={handleFileSelect} aria-label="Edit">
        <span className="fa-layers fa-fw">
            <FontAwesomeIcon icon={faImage} size="lg"/>
            <FontAwesomeIcon icon={faPlus} size="xs" transform="down-13 left-16"/>
        </span>
    </Button>);
}

export default React.memo(FileChooserButton);