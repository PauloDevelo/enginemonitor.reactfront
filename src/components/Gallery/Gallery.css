.action-button {
    background-color: transparent;
    color: white;
    border: 0;
    opacity: 0.5;
}

.action-button:hover {
    background-color: transparent;
    color: white;
    border: 0;
    opacity: 1;
}

.gallery{
    min-height: 58px;
}