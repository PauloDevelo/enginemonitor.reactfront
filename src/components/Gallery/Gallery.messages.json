

{
    "gallerytitle": {
      "defaultMessage": "Gallery",
      "description": "Gallery title",
      "id": "gallerytitle"
    }
  }
  