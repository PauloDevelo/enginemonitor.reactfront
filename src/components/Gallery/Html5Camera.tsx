import React from 'react';

import Camera, { FACING_MODES, IMAGE_TYPES } from 'react-html5-camera-photo';
import 'react-html5-camera-photo/build/css/index.css';
import "./Html5Camera.css"

import {resizeAndSaveBase64Image} from '../../helpers/ImageHelper';

import { ImageModel } from '../../types/Types';

import errorService from '../../services/ErrorService';

type Props = {
    imageParentUiId: string,
    addImage: (image: ImageModel) => void;
    close: ()=> void;
}

const Html5Camera = ({imageParentUiId, addImage, close}: Props) => {

    const onCameraError = (error: any) => {
        close();
    }
 
    const onCameraStart = (stream: MediaStream) => {
    }
 
    const onCameraStop = () => {
        close();
    }

    const onTakePhoto = async (dataUri: string, method: string) => {
        try{
            if(method === "multer"){
                const newImage = await resizeAndSaveBase64Image(dataUri, imageParentUiId);
                addImage(newImage);
            }
        }
        catch(error){
            errorService.addError(error);
        }
    }
 
    return (
        <div id={"cameraContainer"} onClick={close}>
            <Camera
                onTakePhoto = { (dataUri: string) => {
                    onTakePhoto(dataUri, "multer");
                    close();
                }}
                onCameraError = { (error: any) => onCameraError(error) }
                idealFacingMode = {FACING_MODES.ENVIRONMENT}
                idealResolution = {{width: 1024, height: 1024}}
                imageType = {IMAGE_TYPES.JPG}
                imageCompression = {0.92}
                isMaxResolution = {false}
                isImageMirror = {false}
                isSilentMode = {false}
                isDisplayStartCameraError = {true}
                isFullscreen = {false}
                sizeFactor = {1}
                onCameraStart = { (stream: MediaStream) => onCameraStart(stream) }
                onCameraStop = { () => onCameraStop() }
            />
        </div>
    );
}

export default React.memo(Html5Camera);