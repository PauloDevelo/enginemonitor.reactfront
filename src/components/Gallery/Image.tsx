import React from 'react';
import { Media } from 'reactstrap';

import { ImageModel } from '../../types/Types';
import './Image.css';

type Props = {
    image: ImageModel;
    onClickImage: undefined | (() => void);
}
function Image({image, onClickImage}: Props){
    const displayImage = () => {
        if (onClickImage !== undefined){
            onClickImage();
        }
    }

    const alt  = image.title + " - " + image.description;

    return <Media object src={ image.thumbnailUrl } alt={alt} onClick={displayImage} className={"thumbnail grow"}/>
}

export default React.memo(Image);