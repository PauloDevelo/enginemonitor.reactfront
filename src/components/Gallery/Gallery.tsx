import React, { useState, useEffect, useCallback,Fragment } from 'react';
import { Label, Button } from 'reactstrap';

import Lightbox from 'react-image-lightbox';
import 'react-image-lightbox/style.css';

import Html5Camera from "./Html5Camera";

import Image from './Image';

import { ImageModel } from '../../types/Types';
import FileChooserButton from './FileChooserButton';

import imageProxy from '../../services/ImageProxy';
import {resizeAndSaveImage, resizeAndSaveBase64Image} from '../../helpers/ImageHelper';

import { faCamera,faTrashAlt } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

type Props = {
    parentUiId: string
}

function Gallery({parentUiId}: Props){
    const [images, setImages] = useState<ImageModel[]>([]);
    const [isOpen, setOpen] = useState(false);
    const [isCameraOn, setCamera] = useState(false);
    const [index, setIndex] = useState(-1);

    useEffect(() => {
        imageProxy.fetchImages(parentUiId)
        .then(images => {
            setImages(images);
        })
        .catch((error) => {
            console.error("Error when fetching the image for " + parentUiId);
            console.error(error);
        });
    }, [parentUiId]);

    const onCapture = (imageBase64: string) => {
        uploadBase64Image(imageBase64, "multer");
    };

    const uploadBase64Image = async(imageBase64: string, method: string) => {
        if(method === "multer"){
            const newImage = await resizeAndSaveBase64Image(imageBase64, parentUiId);
            const newImages = images.concat(newImage);

            setImages(newImages);
        }
    };

    const onSelectFile = async(file: File) => {
        uploadImageFile(file, "multer")
    };

    const uploadImageFile = async(file: File, method: string) => {
        if(method === "multer"){
            const newImage = await resizeAndSaveImage(file, parentUiId);
            const newImages = images.concat(newImage);

            setImages(newImages);
        }
    };

    const onClickThumbnail = (image:ImageModel, index: number) => {
        setIndex(index);
        setOpen(true);
    }

    const thumbnails = images.map((image, index) => {
        return <Image key={image._uiId} image={image} index={index} onClickImage={onClickThumbnail} />;
    });

    const deleteCurrentImage = useCallback(() => {
        console.log("delete image");
        imageProxy.deleteImage(images[index]).then(deletedImage => {
            const newImages = images.filter(image => image._uiId !== deletedImage._uiId);
            if(newImages.length === 0){
                setOpen(false);
            }
            else{
                setIndex(0);
            }
            setImages(newImages);
        });
    }, [images, index]);

    const additionalActions  =  [
        <Button onClick={deleteCurrentImage} ><FontAwesomeIcon icon={faTrashAlt} size="lg"/></Button>,
    ];

	return(
        <Fragment>
            <div>
                <Label className="font-weight-bold">Gallery image</Label>
                <div className="p-1 border border-secondary rounded shadow">
                    <FileChooserButton onFileSelect={onSelectFile} className="float-right"/>
                    <Button onClick={() => setCamera(true)} className="float-right"><FontAwesomeIcon icon={faCamera} size="lg"/></Button>
                    {thumbnails}
                </div>
            </div>
            {isOpen && (
                <Lightbox
                mainSrc={images[index].url}
                nextSrc={images[(index + 1) % images.length].url}
                prevSrc={images[(index + images.length - 1) % images.length].url}
                mainSrcThumbnail= {images[index].thumbnailUrl}
                nextSrcThumbnail= {images[(index + 1) % images.length].thumbnailUrl}
                prevSrcThumbnail= {images[(index + images.length - 1) % images.length].thumbnailUrl}
                onCloseRequest={() => setOpen(false)}
                onMovePrevRequest={() => setIndex((index + images.length - 1) % images.length)}
                onMoveNextRequest={() => setIndex((index + 1) % images.length) }
                toolbarButtons={additionalActions}
                />
            )}

            {isCameraOn && <Html5Camera close={() => setCamera(false)} onTakePhoto={onCapture}/>}
        </Fragment>
	);
}

export default React.memo(Gallery);