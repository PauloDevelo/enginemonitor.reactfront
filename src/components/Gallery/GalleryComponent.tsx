import React from 'react';
import { Label, Button } from 'reactstrap';

import Image from './Image';

import { ImageModel } from '../../types/Types';
import FileChooserButton from './FileChooserButton';

import { faCamera, faPlus } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import "./Gallery.css";

type Props = {
    images: ImageModel[],
    onClickThumbnail: (index: number) => void,
    onSelectFile: (file: File) => void,
    turnOnCamera: () => void
};

function GalleryComponent({ images, onClickThumbnail, onSelectFile, turnOnCamera }: Props){
    const thumbnails = images.map((image, index) => {
        return <Image key={image._uiId} image={image} onClickImage={() => onClickThumbnail(index)} />;
    });

	return(
        <div>
            <Label className="font-weight-bold">Gallery image</Label>
            <div className="p-1 border border-secondary rounded shadow gallery">
                <FileChooserButton onFileSelect={onSelectFile} className="float-right"/>
                <Button color="light" size="lg" onClick={turnOnCamera} className="float-right">
                    <span className="fa-layers fa-fw">
                        <FontAwesomeIcon icon={faCamera} size="lg"/>
                        <FontAwesomeIcon icon={faPlus} size="xs" transform="down-13 left-16"/>
                    </span>
                </Button>
                {thumbnails}
            </div>
        </div>
	);
}

export default React.memo(GalleryComponent);