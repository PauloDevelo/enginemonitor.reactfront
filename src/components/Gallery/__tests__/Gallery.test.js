import ignoredMessages from '../../../testHelpers/MockConsole';

import React from 'react';
import { mount } from 'enzyme';

import imageProxy from '../../../services/ImageProxy';
import Gallery from '../Gallery';

jest.mock('../../../services/ImageProxy');

describe('Component Gallery', () =>{
    beforeAll(() => {
        ignoredMessages.length = 0;
        ignoredMessages.push("test was not wrapped in act(...)");
    });

    it('should render a list of image using the thumbnail url', () => {
        // Arrange
        const parentUiId = "equipment_01";

        const images = [
            {   name: "DJI_1645.JPG",
                parentUiId: parentUiId,
                sizeInByte: 262985,
                thumbnailUrl: "http://localhost:8000/api/uploads/5d162f2429b0dc049c9c700e/1571876339198thumbnail_DJI_1645.JPG.jpeg",
                url: "http://localhost:8000/api/uploads/5d162f2429b0dc049c9c700e/1571876339188DJI_1645.JPG.jpeg",
                _uiId: "dfc66510-f5f3-11e9-b3b1-f7086d79fbca"
            },
            {
                name: "DSC_0016.JPG",
                parentUiId: parentUiId,
                sizeInByte: 325323,
                thumbnailUrl: "http://localhost:8000/api/uploads/5d162f2429b0dc049c9c700e/1571876358072thumbnail_DSC_0016.JPG.jpeg",
                url: "http://localhost:8000/api/uploads/5d162f2429b0dc049c9c700e/1571876358061DSC_0016.JPG.jpeg",
                _uiId: "eb074110-f5f3-11e9-b3b1-f7086d79fbca"
            },
            {
                name: "DSC_0005.JPG",
                parentUiId: parentUiId,
                sizeInByte: 239719,
                thumbnailUrl: "http://localhost:8000/api/uploads/5d162f2429b0dc049c9c700e/1571876420022thumbnail_DSC_0005.JPG.jpeg",
                url: "http://localhost:8000/api/uploads/5d162f2429b0dc049c9c700e/1571876420020DSC_0005.JPG.jpeg",
                _uiId: "0ff4fd50-f5f4-11e9-b3b1-f7086d79fbca"
            },
        ];

        imageProxy.fetchImages.mockResolvedValue(images);
        
        // Act
        const wrapper = mount(<Gallery parentUiId={parentUiId} />);
        
        // Assert
        return Promise
            .resolve(wrapper)
            .then(() => {
                wrapper.update();

                expect(wrapper).toMatchSnapshot();
            });
    });
});