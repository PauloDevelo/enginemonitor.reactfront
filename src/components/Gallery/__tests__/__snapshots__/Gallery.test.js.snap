// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Gallery should render a list of image using the thumbnail url 1`] = `
<Memo(Gallery)
  parentUiId="equipment_01"
>
  <Memo(GalleryComponent)
    addImage={[Function]}
    images={
      Array [
        Object {
          "_uiId": "dfc66510-f5f3-11e9-b3b1-f7086d79fbca",
          "name": "DJI_1645.JPG",
          "parentUiId": "equipment_01",
          "sizeInByte": 262985,
          "thumbnailUrl": "http://localhost:8000/api/uploads/5d162f2429b0dc049c9c700e/1571876339198thumbnail_DJI_1645.JPG.jpeg",
          "url": "http://localhost:8000/api/uploads/5d162f2429b0dc049c9c700e/1571876339188DJI_1645.JPG.jpeg",
        },
        Object {
          "_uiId": "eb074110-f5f3-11e9-b3b1-f7086d79fbca",
          "name": "DSC_0016.JPG",
          "parentUiId": "equipment_01",
          "sizeInByte": 325323,
          "thumbnailUrl": "http://localhost:8000/api/uploads/5d162f2429b0dc049c9c700e/1571876358072thumbnail_DSC_0016.JPG.jpeg",
          "url": "http://localhost:8000/api/uploads/5d162f2429b0dc049c9c700e/1571876358061DSC_0016.JPG.jpeg",
        },
        Object {
          "_uiId": "0ff4fd50-f5f4-11e9-b3b1-f7086d79fbca",
          "name": "DSC_0005.JPG",
          "parentUiId": "equipment_01",
          "sizeInByte": 239719,
          "thumbnailUrl": "http://localhost:8000/api/uploads/5d162f2429b0dc049c9c700e/1571876420022thumbnail_DSC_0005.JPG.jpeg",
          "url": "http://localhost:8000/api/uploads/5d162f2429b0dc049c9c700e/1571876420020DSC_0005.JPG.jpeg",
        },
      ]
    }
    onClickThumbnail={[Function]}
    parentUiId="equipment_01"
    turnOnCamera={[Function]}
  >
    <div>
      <Label
        className="font-weight-bold"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <label
          className="font-weight-bold"
        >
          Gallery image
        </label>
      </Label>
      <div
        className="p-1 border border-secondary rounded shadow gallery"
      >
        <Memo(AddImageFileButton)
          addImage={[Function]}
          className="float-right"
          parentUiId="equipment_01"
        >
          <Button
            aria-label="Edit"
            className="float-right"
            color="light"
            onClick={[Function]}
            size="lg"
            tag="button"
          >
            <button
              aria-label="Edit"
              className="float-right btn btn-light btn-lg"
              onClick={[Function]}
              type="button"
            >
              <span
                className="fa-layers fa-fw"
              >
                <FontAwesomeIcon
                  border={false}
                  className=""
                  fixedWidth={false}
                  flip={null}
                  icon={
                    Object {
                      "icon": Array [
                        512,
                        512,
                        Array [],
                        "f03e",
                        "M464 448H48c-26.51 0-48-21.49-48-48V112c0-26.51 21.49-48 48-48h416c26.51 0 48 21.49 48 48v288c0 26.51-21.49 48-48 48zM112 120c-30.928 0-56 25.072-56 56s25.072 56 56 56 56-25.072 56-56-25.072-56-56-56zM64 384h384V272l-87.515-87.515c-4.686-4.686-12.284-4.686-16.971 0L208 320l-55.515-55.515c-4.686-4.686-12.284-4.686-16.971 0L64 336v48z",
                      ],
                      "iconName": "image",
                      "prefix": "fas",
                    }
                  }
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size="lg"
                  spin={false}
                  swapOpacity={false}
                  symbol={false}
                  title=""
                  transform={null}
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-image fa-w-16 fa-lg "
                    data-icon="image"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style={Object {}}
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M464 448H48c-26.51 0-48-21.49-48-48V112c0-26.51 21.49-48 48-48h416c26.51 0 48 21.49 48 48v288c0 26.51-21.49 48-48 48zM112 120c-30.928 0-56 25.072-56 56s25.072 56 56 56 56-25.072 56-56-25.072-56-56-56zM64 384h384V272l-87.515-87.515c-4.686-4.686-12.284-4.686-16.971 0L208 320l-55.515-55.515c-4.686-4.686-12.284-4.686-16.971 0L64 336v48z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                </FontAwesomeIcon>
                <FontAwesomeIcon
                  border={false}
                  className=""
                  fixedWidth={false}
                  flip={null}
                  icon={
                    Object {
                      "icon": Array [
                        448,
                        512,
                        Array [],
                        "f067",
                        "M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z",
                      ],
                      "iconName": "plus",
                      "prefix": "fas",
                    }
                  }
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size="xs"
                  spin={false}
                  swapOpacity={false}
                  symbol={false}
                  title=""
                  transform="down-13 left-16"
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-plus fa-w-14 fa-xs "
                    data-icon="plus"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style={
                      Object {
                        "transformOrigin": "-0.5625em 1.3125em",
                      }
                    }
                    viewBox="0 0 448 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      style={Object {}}
                      transform="translate(224 256)"
                    >
                      <g
                        style={Object {}}
                        transform="translate(-512, 416)  scale(1, 1)  rotate(0 0 0)"
                      >
                        <path
                          d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                          fill="currentColor"
                          style={Object {}}
                          transform="translate(-224 -256)"
                        />
                      </g>
                    </g>
                  </svg>
                </FontAwesomeIcon>
              </span>
            </button>
          </Button>
        </Memo(AddImageFileButton)>
        <Button
          className="float-right"
          color="light"
          onClick={[Function]}
          size="lg"
          tag="button"
        >
          <button
            aria-label={null}
            className="float-right btn btn-light btn-lg"
            onClick={[Function]}
            type="button"
          >
            <span
              className="fa-layers fa-fw"
            >
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon={
                  Object {
                    "icon": Array [
                      512,
                      512,
                      Array [],
                      "f030",
                      "M512 144v288c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V144c0-26.5 21.5-48 48-48h88l12.3-32.9c7-18.7 24.9-31.1 44.9-31.1h125.5c20 0 37.9 12.4 44.9 31.1L376 96h88c26.5 0 48 21.5 48 48zM376 288c0-66.2-53.8-120-120-120s-120 53.8-120 120 53.8 120 120 120 120-53.8 120-120zm-32 0c0 48.5-39.5 88-88 88s-88-39.5-88-88 39.5-88 88-88 88 39.5 88 88z",
                    ],
                    "iconName": "camera",
                    "prefix": "fas",
                  }
                }
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size="lg"
                spin={false}
                swapOpacity={false}
                symbol={false}
                title=""
                transform={null}
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-camera fa-w-16 fa-lg "
                  data-icon="camera"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M512 144v288c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V144c0-26.5 21.5-48 48-48h88l12.3-32.9c7-18.7 24.9-31.1 44.9-31.1h125.5c20 0 37.9 12.4 44.9 31.1L376 96h88c26.5 0 48 21.5 48 48zM376 288c0-66.2-53.8-120-120-120s-120 53.8-120 120 53.8 120 120 120 120-53.8 120-120zm-32 0c0 48.5-39.5 88-88 88s-88-39.5-88-88 39.5-88 88-88 88 39.5 88 88z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </FontAwesomeIcon>
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon={
                  Object {
                    "icon": Array [
                      448,
                      512,
                      Array [],
                      "f067",
                      "M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z",
                    ],
                    "iconName": "plus",
                    "prefix": "fas",
                  }
                }
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size="xs"
                spin={false}
                swapOpacity={false}
                symbol={false}
                title=""
                transform="down-13 left-16"
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-plus fa-w-14 fa-xs "
                  data-icon="plus"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={
                    Object {
                      "transformOrigin": "-0.5625em 1.3125em",
                    }
                  }
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    style={Object {}}
                    transform="translate(224 256)"
                  >
                    <g
                      style={Object {}}
                      transform="translate(-512, 416)  scale(1, 1)  rotate(0 0 0)"
                    >
                      <path
                        d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                        fill="currentColor"
                        style={Object {}}
                        transform="translate(-224 -256)"
                      />
                    </g>
                  </g>
                </svg>
              </FontAwesomeIcon>
            </span>
          </button>
        </Button>
        <Memo(Thumbnail)
          image={
            Object {
              "_uiId": "dfc66510-f5f3-11e9-b3b1-f7086d79fbca",
              "name": "DJI_1645.JPG",
              "parentUiId": "equipment_01",
              "sizeInByte": 262985,
              "thumbnailUrl": "http://localhost:8000/api/uploads/5d162f2429b0dc049c9c700e/1571876339198thumbnail_DJI_1645.JPG.jpeg",
              "url": "http://localhost:8000/api/uploads/5d162f2429b0dc049c9c700e/1571876339188DJI_1645.JPG.jpeg",
            }
          }
          key="dfc66510-f5f3-11e9-b3b1-f7086d79fbca"
          onClickImage={[Function]}
        >
          <Media
            alt="undefined - undefined"
            className="thumbnail grow"
            object={true}
            onClick={[Function]}
            src="http://localhost:8000/api/uploads/5d162f2429b0dc049c9c700e/1571876339198thumbnail_DJI_1645.JPG.jpeg"
          >
            <img
              alt="undefined - undefined"
              className="thumbnail grow media-object"
              onClick={[Function]}
              src="http://localhost:8000/api/uploads/5d162f2429b0dc049c9c700e/1571876339198thumbnail_DJI_1645.JPG.jpeg"
            />
          </Media>
        </Memo(Thumbnail)>
        <Memo(Thumbnail)
          image={
            Object {
              "_uiId": "eb074110-f5f3-11e9-b3b1-f7086d79fbca",
              "name": "DSC_0016.JPG",
              "parentUiId": "equipment_01",
              "sizeInByte": 325323,
              "thumbnailUrl": "http://localhost:8000/api/uploads/5d162f2429b0dc049c9c700e/1571876358072thumbnail_DSC_0016.JPG.jpeg",
              "url": "http://localhost:8000/api/uploads/5d162f2429b0dc049c9c700e/1571876358061DSC_0016.JPG.jpeg",
            }
          }
          key="eb074110-f5f3-11e9-b3b1-f7086d79fbca"
          onClickImage={[Function]}
        >
          <Media
            alt="undefined - undefined"
            className="thumbnail grow"
            object={true}
            onClick={[Function]}
            src="http://localhost:8000/api/uploads/5d162f2429b0dc049c9c700e/1571876358072thumbnail_DSC_0016.JPG.jpeg"
          >
            <img
              alt="undefined - undefined"
              className="thumbnail grow media-object"
              onClick={[Function]}
              src="http://localhost:8000/api/uploads/5d162f2429b0dc049c9c700e/1571876358072thumbnail_DSC_0016.JPG.jpeg"
            />
          </Media>
        </Memo(Thumbnail)>
        <Memo(Thumbnail)
          image={
            Object {
              "_uiId": "0ff4fd50-f5f4-11e9-b3b1-f7086d79fbca",
              "name": "DSC_0005.JPG",
              "parentUiId": "equipment_01",
              "sizeInByte": 239719,
              "thumbnailUrl": "http://localhost:8000/api/uploads/5d162f2429b0dc049c9c700e/1571876420022thumbnail_DSC_0005.JPG.jpeg",
              "url": "http://localhost:8000/api/uploads/5d162f2429b0dc049c9c700e/1571876420020DSC_0005.JPG.jpeg",
            }
          }
          key="0ff4fd50-f5f4-11e9-b3b1-f7086d79fbca"
          onClickImage={[Function]}
        >
          <Media
            alt="undefined - undefined"
            className="thumbnail grow"
            object={true}
            onClick={[Function]}
            src="http://localhost:8000/api/uploads/5d162f2429b0dc049c9c700e/1571876420022thumbnail_DSC_0005.JPG.jpeg"
          >
            <img
              alt="undefined - undefined"
              className="thumbnail grow media-object"
              onClick={[Function]}
              src="http://localhost:8000/api/uploads/5d162f2429b0dc049c9c700e/1571876420022thumbnail_DSC_0005.JPG.jpeg"
            />
          </Media>
        </Memo(Thumbnail)>
      </div>
    </div>
  </Memo(GalleryComponent)>
  <ModalEditImage
    onImageDeleted={[Function]}
    onImageSaved={[Function]}
    toggle={[Function]}
    visible={false}
  >
    <CSSTransition
      classNames="modal"
      in={false}
      timeout={300}
    >
      <Transition
        appear={false}
        enter={true}
        exit={true}
        in={false}
        mountOnEnter={false}
        onEnter={[Function]}
        onEntered={[Function]}
        onEntering={[Function]}
        onExit={[Function]}
        onExited={[Function]}
        onExiting={[Function]}
        timeout={300}
        unmountOnExit={false}
      >
        <Modal
          autoFocus={true}
          backdrop={true}
          backdropTransition={
            Object {
              "mountOnEnter": true,
              "timeout": 150,
            }
          }
          centered={false}
          fade={false}
          isOpen={false}
          keyboard={true}
          modalTransition={
            Object {
              "timeout": 300,
            }
          }
          onClosed={[Function]}
          onOpened={[Function]}
          returnFocusAfterClose={true}
          role="dialog"
          scrollable={false}
          toggle={[Function]}
          unmountOnClose={true}
          zIndex={1050}
        />
      </Transition>
    </CSSTransition>
    <ModalYesNoConfimation
      className="modal-dialog-centered"
      isActing={false}
      message={
        Object {
          "defaultMessage": "Are you sure you want to delete this image?",
          "description": "confirmation message",
          "id": "imageDeleteMessage",
        }
      }
      no={[Function]}
      title={
        Object {
          "defaultMessage": "Delete image",
          "description": "modal title",
          "id": "imageDeleteTitle",
        }
      }
      toggle={[Function]}
      visible={false}
      yes={[Function]}
    >
      <CSSTransition
        classNames="modal"
        in={false}
        timeout={300}
      >
        <Transition
          appear={false}
          enter={true}
          exit={true}
          in={false}
          mountOnEnter={false}
          onEnter={[Function]}
          onEntered={[Function]}
          onEntering={[Function]}
          onExit={[Function]}
          onExited={[Function]}
          onExiting={[Function]}
          timeout={300}
          unmountOnExit={false}
        >
          <Modal
            autoFocus={true}
            backdrop={true}
            backdropTransition={
              Object {
                "mountOnEnter": true,
                "timeout": 150,
              }
            }
            centered={false}
            className="modal-dialog-centered"
            fade={false}
            isOpen={false}
            keyboard={true}
            modalTransition={
              Object {
                "timeout": 300,
              }
            }
            onClosed={[Function]}
            onOpened={[Function]}
            returnFocusAfterClose={true}
            role="dialog"
            scrollable={false}
            toggle={[Function]}
            unmountOnClose={true}
            zIndex={1050}
          />
        </Transition>
      </CSSTransition>
    </ModalYesNoConfimation>
  </ModalEditImage>
</Memo(Gallery)>
`;
